### ISSUE TITLE
Reproduced result slightly lower than the reported.

### ISSUE THREAD

**Issue by junwenchen** [amazon-research/long-short-term-transformer#5]
Hi,

I have tried the code and get 68.3% on thumos dataset, which is slightly lower than the result on paper (69.5%)

Can you provide one or two features and label file so that I can compare with and have sanity check?

**Comment by xumingze0308** [comment]
Hi,

Sorry that we cannot release anything related to data. Getting results lower than 69.0% is usually caused by using different FPSs or input sizes. Results from 69.3% to 70.0% are normal on different GPUs or machines. Here are something you can try:

First, make sure the RGB frames and optical flows are computed under 24 FPS. They should keep their original aspect ratio, 320x180. The chunk size is 6 (the video is converted to 4 FPS).

Second, the labels are generated in chunk level. We convert the ground truth from seconds to 4 FPS, and label each chunk accordingly, where we used floor() for action start and ceil() for action end.

Third, the RGB features are extracted by the 4-th frame in our implementation. The optical flow should be computed using denseflow with -b=20 and -a=tvl1. When extracting the optical flow features using BNInception, you should follow https://github.com/yjxiong/action-detection/blob/master/transforms.py and use the multiple crops.

For feature extraction, make sure to use the checkpoint we pointed in the README instruction.

**Comment by priyamdey** [comment]
Hi. I needed some more clarifications on your above response:
1. What transformation(s) do you perform on the RGB frames before passing it to RN50 (resize to 256 & center crop to 224?)? Is it the same for TVSeries dataset as well, coz TVSeries has frames of size 716x576.
2. Just to confirm, do you use ```GroupRandomCrop``` from [here](https://github.com/yjxiong/action-detection/blob/master/transforms.py) for multi-cropping of flow features or a combination of transforms? And is the flow crop size 224x224?

Thanks

**Comment by xumingze0308** [comment]
1. We only normalized the RGB frames with ImageNet mean and std. We don't need to resize them to 256 and then crop to 224, since ResNets can handle inputs with arbitrary size. You just need to average pool the last feature from H' x W' x C to C.

2. Yes. To clarify there, only extracting features with BNInception needs multi-crop, as it can only receive inputs of size 224x224.

**Comment by priyamdey** [comment]
One more thing. Do you do any normalization of the flow fields before sending it to BNInception? If yes, how?

**Comment by xumingze0308** [comment]
Yes, and we only normalize the input flow fields by subtracting the mean (128) for each x and y.

**Comment by Prot-debug** [comment]
@priyamdey 
Hi priyamdey，

I followed the `url` [https://github.com/yjxiong/action-detection/blob/master/transforms.py](url) that author mentionded in this issue. But I found that this data preprocessing is so inefficient, that is, processing one `.jpg` files frame by frame. And it's easy to cause memory leak problems by using the function `PIL.Image.open()`.
I don't know where I am going wrong, my code process is as follows. Could you give me some advice plz?

```
def transforms_img(img_list):

    trans = torchvision.transforms.Compose([
        GroupScale(256),
        GroupRandomCrop(224),
        Stack(),
        ToTorchFormatTensor(),
        GroupNormalize(
            mean=[128],
            std=[128]
        )]
    )

    for i, img_dir in enumerate(img_list):
        with open(img_dir, 'rb') as open_file:
            img = Image.open(open_file).convert('L')
            color_group = [img]
            rst = trans(color_group)
            del color_group
            del img
        if i == 0:
            stack_img = rst
            del rst
        else:
            stack_img = torch.cat((stack_img, rst), dim=0)
            del rst
    gc.collect() 

```
